plugins/vpc-branch-eni/plugin/commands.go

```diff
@@ -296,6 +295,22 @@ func (plugin *Plugin) createVLANLink(branch *eni.Branch, linkName string, ipAddr
 	return nil
 }
 
+func getGatewayIPAddress(ipAddress *net.IPNet, gatewayIPAddress net.IP) (net.IP, error) {
```

Comment:
That looks great. Just one more thing: Should we move this function to netconfig.go? There are two structs there: NetConfig and NetConfigJSON. NetConfigJSON is the raw JSON file contents. NetConfig is the actual configuration, compiled after all the config decisions are made, that is passed to command handlers for execution. What we are doing here is basically a configuration decision, and I think that should be handled as an else case to the if block at line 159 in netconfig.go. This way, the default gateway will be known to all code in the plugin. Command handlers don't make a decision, they just execute the configuration. You can then move your test cases to netconfig_test.go.

Reply:
that makes sense. updated